nevergrad/optimization/optimizerlib.py

```diff
@@ -137,10 +137,14 @@ def _internal_ask_candidate(self) -> p.Parameter:
                 data = mutator.portfolio_discrete_mutation(pessimistic_data, max(1, int(self._adaptive_mr * self.dimension)))
             elif mutation == "discreteBSO":
                 assert self.budget is not None, "DiscreteBSO needs a budget."
-                intensity: int = int(self.dimension - self._num_ask * self.dimension / self.budget)
+                intensity = int(self.dimension - self._num_ask * self.dimension / self.budget)
                 if intensity < 1:
                     intensity = 1
                 data = mutator.portfolio_discrete_mutation(pessimistic_data, intensity)
+            elif mutation == "lengler":
+                alpha = 1.54468
+                intensity = float(max(1.0, self.dimension * (alpha * np.log(self.num_ask) / self.num_ask)))
```

Comment:
@teytaud should it be an integer? `portfolio_discrete_mutation` only takes integers (should that be modified?)

Reply:
I'll use int (as you did) for now as maybe this is more meaningful as it is somehow homogeneous to a number of variables, but given the stochastic nature of the method we could discuss this...

Reply:
I used integer because you used integer initially :p it's definitely up to you, as long as it is coherent